File: app-mvp/src/main/java/xuk/imaginary/gui/gank/show/GankContract.java
package xuk.imaginary.gui.gank.show;

import xuk.imaginary.BasePresenter;
import xuk.imaginary.BaseView;
import xuk.imaginary.data.GankBean;

/**
 * This specifies the contract between the view and the presenter.
 *
 * @author xuie
 */
public interface GankContract {

    interface View extends BaseView<Presenter> {

        void refresh(GankBean gb);
    }

    interface Presenter extends BasePresenter {
        void getGank(int year, int month, int day);

        void getGank(String date);
    }
}

File: app-mvp-kotlin/src/main/java/xuk/imaginary/gui/gank/show/GankContract.kt
package xuk.imaginary.gui.gank.show

import xuk.imaginary.BasePresenter
import xuk.imaginary.BaseView
import xuk.imaginary.data.GankBean

/**
 * This specifies the contract between the view and the presenter.
 */
interface GankContract {

  interface View : BaseView<Presenter> {

    fun refresh(gb: GankBean)
  }

  interface Presenter : BasePresenter {
    fun getGank(year: Int, month: Int, day: Int)

    fun getGank(date: String)
  }
}
